src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -839,17 +839,68 @@ int wm_vuldet_create_file(const char *path, const char *source) {
     const char *ROOT = "root";
     const char *sql;
     const char *tail;
-    sqlite3 *db = NULL;
-    sqlite3_stmt *stmt = NULL;
     int result;
     uid_t uid;
     gid_t gid;
+    static bool already_removed = false;
+    bool db_existed = true;
+    sqlite3 *db = NULL;
+    sqlite3_stmt *stmt = NULL;
 
     // Open the database file, or create it
-    if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL)) {
-        mterror(WM_VULNDETECTOR_LOGTAG, VU_CREATE_DB_ERROR);
-        return wm_vuldet_sql_error(db, stmt);
+    if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE, NULL)) {
+        db_existed = false;
+        if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL)) {
+            mterror(WM_VULNDETECTOR_LOGTAG, VU_CREATE_DB_ERROR);
+            return wm_vuldet_sql_error(db, stmt);
+        }
+    }
+
+    // Always try to create the metadata table
+    mtdebug2(WM_VULNDETECTOR_LOGTAG, "Force create VU_METADATA if not exists");
+    sqlite3_exec(db, vu_queries[CREATE_VU_METADATA], NULL, NULL, NULL);
+
```

Comment:
To avoid having to modify both the schema and the global variable, wouldn't it be better to update the value after removing the DB?

It could be something like the following:
```suggestion

    if (already_removed) {
        if (wm_vuldet_prepare(db, vu_queries[UPDATE_VU_METADATA], -1, &stmt, NULL) != SQLITE_OK) {
            return wm_vuldet_sql_error(db, stmt);
        }
        sqlite3_bind_text(stmt, 1, CVE_DB_VERSION, -1, NULL);

        if (wm_vuldet_step(stmt) != SQLITE_DONE) {
            return wm_vuldet_sql_error(db, stmt);
        }
        wdb_finalize(stmt);
    }

```

Reply:
Well, I'd rather simplify the upgrade than not do it at all because the code is already tested.
Because I don't see it as unnecessary, but in my opinion it would be a more sustainable code.

Reply:
Reviewing this suggestion again, I understand that this code added as suggested here would imply the elimination of part of the current code, if so, what would be the code to eliminate?

Reply:
It does not imply the removal of any code, it is simply an addition to update the schema versioning to the same version set in the `CVE_DB_VERSION` variable in case the DB has been updated.

In any case, if you add this change, you must also add the corresponding query: `UPDATE_VU_METADATA`.

Reply:
Knowing this, then the code section that goes from L864-L879 would be duplicated.